Formula/apache-pulsar.rb

```diff
@@ -0,0 +1,83 @@
+class ApachePulsar < Formula
+  desc "Cloud-native distributed messaging and streaming platform"
+  homepage "https://pulsar.apache.org/"
+  url "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.8.0/apache-pulsar-2.8.0-src.tar.gz"
+  mirror "https://archive.apache.org/dist/pulsar/pulsar-2.8.0/apache-pulsar-2.8.0-src.tar.gz"
+  sha256 "0e161a81c62c7234c1e0c243bb6fe30046ec1cd01472618573ecdc2a73b1163b"
+  license "Apache-2.0"
+  head "https://github.com/apache/pulsar.git", branch: "master"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cppunit" => :build
+  depends_on "libtool" => :build
+  depends_on "maven" => :build
+  depends_on "pkg-config" => :build
+  depends_on "protobuf" => :build
+  depends_on arch: :x86_64
+  depends_on "openjdk@11"
+
+  def install
+    chmod "+x", "src/rename-netty-native-libs.sh"
+    with_env(
+      "PATH"      => "#{Formula["openjdk@11"].bin}:#{ENV["PATH"]}",
+      "JAVA_HOME" => Formula["openjdk@11"].opt_prefix,
+      "TMPDIR"    => buildpath,
+    ) do
+      system(
+        "mvn",
+        "-X",
+        "clean",
+        "package",
+        "-DskipTests",
+        "-Pcore-modules",
+      )
+    end
+    built_version = if build.head?
+      `python src/get-project-version.py`.strip
+    else
+      version
+    end
+    binpfx = "apache-pulsar-#{built_version}"
+    system "tar", "-xf", "distribution/server/target/#{binpfx}-bin.tar.gz"
+    libexec.install binpfx+"/bin", binpfx+"/lib", binpfx+"/instances", binpfx+"/conf"
+    (libexec/"lib/presto/bin/procname/Linux-ppc64le").rmtree
+    share.install binpfx+"/examples"
+    share.install binpfx+"/licenses"
+    (var/"log/pulsar").mkpath
+    (etc/"pulsar").mkpath
+    (etc/"pulsar").install_symlink libexec/"conf"
+
+    Pathname.glob("#{libexec}/bin/*") do |path|
+      if !path.fnmatch?("*common.sh") && !path.directory?
+        bin_name = path.basename
+        (bin+bin_name).write <<~EOS
+          #!/bin/bash
+          export PATH="#{Formula["openjdk@11"].bin}:$PATH"
+          export JAVA_HOME="#{Formula["openjdk@11"].opt_prefix}"
+          exec "#{libexec}/bin/#{bin_name}" "$@"
+        EOS
```

Comment:
This is doing essentially the same as the `overridable_java_home_env` wrapper we have so we should probably use that one.

Reply:
👍, addressed!

Reply:
As a followup, I switched to `java_home_env`, as tests will not pass on lower or higher versions of openJDK. Pulsar as a project has enough native/weird dependencies that I do not think there is a benefit to allowing overridden JVMs, and that will only lead to bug reports due to folks whose environment variables are causing the usage of an incompatible JVM.

Reply:
Does it download it's own native dependencies? Because that is not supposed to happen, see https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

Reply:
It downloads source code (for native stuff, C/C++/etc.) and builds it; it does not download precompiled binaries.

It also downloads many java bytecode binaries and packages in order to run its build. This is done by every Maven-based project in Homebrew, without exception, which is most java formulae.

Reply:
The java bytecode I'm not too worried about, that's supposed to be portable. The native stuff is more worrying, can we see a list somewhere to ensure we're not compiling things that are already in Homebrew?

Reply:
Netty is the one that was causing me trouble. I don't know enough Maven to generate a list of only compiled dependencies. However, Netty does not seem to be present in Homebrew or any taps I can find.